jdaviz/configs/default/plugins/metadata_viewer/metadata_viewer.py

```diff
@@ -2,13 +2,28 @@
 
 from jdaviz.core.registries import tray_registry
 from jdaviz.core.template_mixin import PluginTemplateMixin, DatasetSelectMixin
+from jdaviz.core.user_api import PluginUserApi
 from jdaviz.utils import PRIHDR_KEY, COMMENTCARD_KEY
 
 __all__ = ['MetadataViewer']
 
 
 @tray_registry('g-metadata-viewer', label="Metadata Viewer")

```

Comment:
should we consider changing the label to just "Metadata"?

Reply:
I'm not sure about just "Metadata" but I agree "Metadata Viewer" is worse. Maybe "Metadata Plugin"?

Reply:
If you remove "Viewer," they might get the wrong idea that they can edit the metadata with this plugin. In IRAF, you could use a task to edit FITS header, so...

Reply:
if I switch exposing the `metadata` traitlet to the read-only infrastructure added in #1643, then maybe that and some docs updates would avoid that confusion?  Is there any chance we'll ever extend this to allow editing metadata (either from the UI or API)?

Reply:
I can see editing metadata as a common use case and worth supporting. Implementing it on the other hand would be a...difficult task.

Reply:
Let's not open that can of worms here. Can you imagine if someone accidentally overwrites a keyword that is important for WCS transformation?

Reply:
I did change it so that `plugin.metadata` is read-only from the user API (otherwise they could change the traitlet and the plugin would update to show that, but it would not persist when changing the data dropdown, etc).

Reply:
With the readonly `plugin.metadata` is anyone opposed to removing "Viewer"?  I just think `viz.plugins['Metadata']` and having "Metadata" in the plugin tray menu feels simpler and more natural, as long as we can sufficiently avoid user-confusion through the API and docs.

Reply:
Yeah, that is a good compromise. Thanks!